src/tests/harness/BenchmarkDotNet.Extensions.Tests/UniqueValuesGeneratorTests.cs

```diff
@@ -72,5 +72,115 @@ public void GetNonDefaultValueReturnsNonDefaultValue()
             Assert.NotEqual(default(int), ValuesGenerator.GetNonDefaultValue<int>());
             Assert.NotEqual(default(string), ValuesGenerator.GetNonDefaultValue<string>());
         }
+
+        [Fact]
+        public void SupportsByte()
+        {
+            Supports<byte>();
+        }
+
+        [Fact]
+        public void SupportsChar()
+        {
+            Supports<char>();
+        }
+
+        [Fact]
+        public void SupportsShort()
+        {
+            Supports<short>();
+        }
+
+        [Fact]
+        public void SupportsUnsignedShort()
+        {
+            Supports<ushort>();
+        }
+
+        [Fact]
+        public void SupportsInteger()
+        {
+            Supports<int>();
+        }
+
+        [Fact]
+        public void SupportsUnsignedInteger()
+        {
+            Supports<uint>();
+        }
+
+        [Fact]
+        public void SupportsLong()
+        {
+            Supports<long>();
+        }
+ 
+        [Fact]
+        public void SupportsUnsignedLong()
+        {
+            Supports<ulong>();
+        }
+
+        [Fact]
+        public void SupportsFloat()
+        {
+            Supports<float>();
+        }
+
+        [Fact]
+        public void SupportsDouble()
+        {
+            Supports<double>();
+        }
+
+        [Fact]
+        public void SupportsBool()
+        {
+            SupportsArray<bool>();
```

Comment:
`bool` is a special case, as there's really no useful answer to the non-default or unique values methods.

Reply:
Turns out that with some tweaks we _can_ reasonably treat bools like any other type. Added tests to ensure we throw if someone asks for too many distinct values (like we do for bytes)